modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManagerComponent.java

```diff
@@ -182,6 +182,14 @@ protected BlobProvider getBlobProvider(String key, String repositoryName) {
      */
     @Override
     public String writeBlob(Blob blob, Document doc, String xpath) throws IOException {
+        if (blob == null) {
+            if (MAIN_BLOB_XPATH.equals(xpath) && doc.isUnderRetentionOrLegalHold()) {
```

Comment:
IIRC it is only about `file:content`, right? Other blobs can be modified if a document is under retention?